### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Bombardier256**

### CODE REVIEW

File: app/battle/contest.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        return knight_instances
+
+    @staticmethod
+    def prepare_for_battle(ritters: list) -> None:
+
+        for ritter in ritters:
+            ritter.prepare_for_battle()
+
+    @classmethod
+    def fight(cls, knight_1: Knight, knight_2: Knight) -> None:
```
Comment (vasylhnatiuk) [inline app/battle/contest.py:31]: why did you use @classmethod?? use static or usual method here

File: app/battle/contest.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        knight_1.hp -= knight_2.power - knight_1.protection
+        knight_2.hp -= knight_1.power - knight_2.protection
+        if knight_1.hp <= 0:
+            knight_1.hp = 0
+
+        if knight_2.hp <= 0:
+            knight_2.hp = 0
+
+    @classmethod
+    def start_battles(cls, ritters: list) -> None:
```
Comment (vasylhnatiuk) [inline app/battle/contest.py:41]: the same here

File: app/knight/knight.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    def apply_armour(self) -> None:
+        for arm in self.armour:
+            self.protection += arm["protection"]
+
+    def apply_weapon(self) -> None:
+        self.power += self.weapon["power"]
+
+    def apply_potion(self) -> None:
+        if self.potion is not None:
```
Comment (vasylhnatiuk) [inline app/knight/knight.py:27]: instead of duplicating use loop and `setattr`

File: app/battle/contest.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            knight_instances.append(Knight(
+                ritters[ritter]["name"],
+                ritters[ritter]["power"],
+                ritters[ritter]["hp"],
+                ritters[ritter]["armour"],
+                ritters[ritter]["weapon"],
+                ritters[ritter]["potion"]
+            ))
+
+        return knight_instances
```
Comment (vasylhnatiuk) [inline app/battle/contest.py:22]: you can save knight_instances as Contest attribute and use it in other functions

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]

File: app/battle/contest.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+        for ritter in knights:
+            self.ritters.append(Knight(
+                knights[ritter]["name"],
+                knights[ritter]["power"],
+                knights[ritter]["hp"],
+                knights[ritter]["armour"],
+                knights[ritter]["weapon"],
+                knights[ritter]["potion"]
+            ))
```
Comment (vsmutok) [inline app/battle/contest.py:20]:
```suggestion
            self.ritters.append(Knight(**knights[ritter]))
```

File: app/battle/contest.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+    @staticmethod
+    def fight(knight_1: Knight, knight_2: Knight) -> None:
+        knight_1.hp -= knight_2.power - knight_1.protection
+        knight_2.hp -= knight_1.power - knight_2.protection
+        if knight_1.hp <= 0:
+            knight_1.hp = 0
+
+        if knight_2.hp <= 0:
+            knight_2.hp = 0
```
Comment (vsmutok) [inline app/battle/contest.py:35]: DRY - try to avoid code duplication

File: app/main.py [hunk 7]
```diff
@@ -1,3 +1,6 @@
+from app.battle.contest import Contest
+
+
 KNIGHTS = {
```
Comment (vsmutok) [inline app/main.py:4]: move dict with all data to separate module

**Review by vsmutok — CHANGES_REQUESTED** [review]

**Pull request closed without merging**